-1

Tested at: 2d2cccf3c90050692a0838d3f5be23d770ddde95

CMSSW: CMSSW_11_1_CLANG_X_2020-04-03-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-547418/5523/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/plugins/GlobalTightHalo2016Filter.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/plugins/BadGlobalMuonTagger.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/plugins/EEBadScFilter.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/plugins/BadParticleFilter.cc 
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/plugins/EcalDeadCellBoundaryEnergyFilter.cc:56:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_CLANG_X_2020-04-03-2300/src/RecoMET/METFilters/interface/EcalBoundaryInfoCalculator.h:229:54: error: no viable conversion from 'std::map<CdOrientation, CdOrientation, std::less<CdOrientation>, std::allocator<std::pair<const CdOrientation, CdOrientation>>>::const_iterator' (aka '_Rb_tree_const_iterator<std::pair<const CdOrientation, CdOrientation>>') to 'std::map<CdOrientation, CdOrientation>::iterator' (aka '_Rb_tree_iterator<std::pair<const CdOrientation, CdOrientation>>')
    std::map<CdOrientation, CdOrientation>::iterator oIt = oppositeDirs.find(currDirection);
                                                     ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../include/c++/8.3.1/bits/stl_tree.h:256:12: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::map<CdOrientation, CdOrientation, std::less<CdOrientation>, std::allocator<std::pair<const CdOrientation, CdOrientation>>>::const_iterator' (aka '_Rb_tree_const_iterator<std::pair<const CdOrientation, CdOrientation>>') to 'const std::_Rb_tree_iterator<std::pair<const CdOrientation, CdOrientation>> &' for 1st argument
    struct _Rb_tree_iterator
           ^
</pre>

--- a/RecoMET/METFilters/interface/EcalBoundaryInfoCalculator.h
+++ b/RecoMET/METFilters/interface/EcalBoundaryInfoCalculator.h
@@ -226,7 +226,7 @@ private:
   }
 
   CdOrientation goBackOneCell(CdOrientation currDirection, EcalDetId prev, CaloNavigator<EcalDetId>* theEcalNav) const {
-    std::map<CdOrientation, CdOrientation>::iterator oIt = oppositeDirs.find(currDirection);
+    auto oIt = oppositeDirs.find(currDirection);
     CdOrientation oppDirection = none;
     if (oIt != oppositeDirs.end()) {
       oppDirection = oIt->second;

